src/history/hs.c

```diff
@@ -165,6 +165,9 @@ __wt_hs_create(WT_SESSION_IMPL *session, const char **cfg)
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /* Drop the lookaside file if it still exists. */
```

Comment:
Why we move the logic here? I think drop las should be outside hs open, otherwise, we will try to drop it every time we call this.

Reply:
@quchenhao The existing code already unconditionally calls `__wt_hs_cleanup_las` followed by `__wt_hs_create`. Therefore, there is no change in logic here. Secondly, this only happens once at startup so there should be no performance penalty in normal operations.

Reply:
I understand there is no logic change here at the moment because `__wt_hs_open` is only called here. My concern is that if we need to call `__wt_hs_open` in some other place, we will accidentally call `__wt_hs_cleanup_las` as well. So I think they should not be coupled together.

Reply:
We can make it optional through a function argument but I am curious if other modules (for the sake of module isolation) should be aware of old LAS functionality?

Also is there a harm in deleting old LAS file unconditionally?

Reply:
I think the las file should only be deleted once at start. Therefore, I think we should not be doing it inside the `__wt_hs_open` function instead we should do it separately as the current code does. However, I can't think of deleting LAS file will do any harm as it is useless across restart.

Reply:
Looking at the code it might not be easy to remember if we have already deleted the LAS file. The config we use for `__wt_schema_drop` ensures that we never get an error if the file doesn't exist.

Can you think of a scenario where the caller might just want to start the HS and not delete LAS file?

Reply:
No, I can't. Like I said, LAS is useless across restart.

Reply:
My preference is to handle LAS as part of hs_open(), kind of compatibility logic. External components need not bother about their relationship i.e. in which order things has to be deleted and opened. My model is earlier cache overflow mechanism is LAS and the current one is history store, so having logic to deal with LAS as part of open is like history store is handling its compatibility bit.

Reply:
@bvpvamsikrishna I had the discussion with @quchenhao and we agreed that `__wt_hs_open` should manage the existing LAS file.